src/Generators/euclideangenerators.jl

```diff
@@ -0,0 +1,48 @@
+"""
+    struct Euclidean <: RandomGenerator
+
+A struct representing a graph generator for a Euclidean graph.
+
```

Comment:
These are the equivalents of `euclidean_graph(...)`.

Reply:
We probably should document what a euclidean graph is. Also this kind of distances seems to allow more kind of distances than just euclidean (by specifying `p`)  so something like `DistanceGraph` seems more appropriate.